Libraries/Map/Map.h

```diff
@@ -0,0 +1,220 @@
+#ifndef Map_h
+#define Map_h
+#include "TVector.h"
+
+class Map {
+  private:
+    uint8_t currRow;
+    uint8_t currColumn;
+    uint8_t numRows;
+    uint8_t numColumns;
+    uint8_t orientation;
+    uint8_t unvisitedTiles;
+    TVector<TVector<Tile> > map;
+    
+  public:
+    Map(Tile initialTile);
+
+    uint8_t getRows();
+    uint8_t getColumns();
+    uint8_t getCurrRow();
+    uint8_t getCurrColumn();
+    uint8_t getOrientation();
+    uint8_t getUnvisitedTiles();
+
+    Tile getTile(const uint8_t row, const uint8_t column);
+    Tile currTile();
+    Tile northTile();
+    Tile eastTile();
+    Tile southTile();
+    Tile westTile();
+
+    void setTile(Tile tile, char position);
+  
+    void addRowFirst(Tile newTile);
+    void addRowLast(Tile newTile);
+    void addColumnFirst(Tile newTile);
+    void addColumnLast(Tile newTile);
+
+    void moveNorth();
+    void moveEast();
+    void moveSouth();
+    void moveWest();
+
+    void print();
+};
+
+Map::Map(Tile initialTile) {
+  currRow = 0;
+  currColumn = 0;
+  numRows = 1;
+  numColumns = 1;
+  orientation = 1;
+  TVector<Tile> row;
+
+  row.pushTop(initialTile);
+  map.pushTop(row);
+}
+
+uint8_t Map::getRows() {
+  return numRows;
+}
+
+uint8_t Map::getColumns() {
+  return numColumns;
+}
+
+uint8_t Map::getCurrRow() {
+  return currRow;
+}
+
+uint8_t Map::getCurrColumn() {
+  return currColumn;
+}
+
+uint8_t Map::getOrientation() {
+  return orientation;
+}
+
+uint8_t Map::getUnvisitedTiles() {
+  unvisitedTiles = 0;
+  for (int row = 0 ; row < numRows ; row++) {
+    for (int column = 0 ; column < numColumns ; column++) {
+      if (map[row][column].getVisited() == false && map[row][column].getAccessible() == true && map[row][column].getBlack() == false) {
+        unvisitedTiles++;
+      }
+    }
+  }
+  return unvisitedTiles;
+}
+
+Tile Map::getTile(const uint8_t row, const uint8_t column) {
+  return map[row][column];
+}
+
+Tile Map::currTile() {
+  return map[currRow][currColumn];
+}
+
+Tile Map::northTile() {
+  return map[currRow - 1][currColumn];
+}
+
+Tile Map::eastTile() {
+  return map[currRow][currColumn + 1];
+}
+
+Tile Map::southTile() {
+  return map[currRow + 1][currColumn];
+}
+
+Tile Map::westTile() {
+  return map[currRow][currColumn - 1];
+}
+
+void Map::setTile(Tile tile, char position) {
+  if (position == '0') map[currRow][currColumn] = tile;
+  else if (position == 'N') map[currRow - 1][currColumn] = tile;
+  else if (position == 'E') map[currRow][currColumn + 1] = tile;
+  else if (position == 'S') map[currRow + 1][currColumn] = tile;
+  else if (position == 'W') map[currRow][currColumn - 1] = tile;
+}
+
+void Map::addRowFirst(Tile newTile) {
+  TVector<Tile> newRow;
+  Tile emptyTile;
+  
+  for(int column = 0 ; column < numColumns ; column++) {
+    newRow.pushTop(emptyTile);
+  }
```

Comment:
Maybe this can be on a function called createEmptyVector.
Or, create a consturctor of TVector, which receives the size and type to replicate. As in:
Tile empty_tile;
TVector<Tile> my_empty_vector(5, empty_tile);

Or something similar.

Reply:
where 5 would be num_columns